### DISCUSSION TITLE
Circular imports with import chain of more than 2 modules are not detected anymore

### DISCUSSION THREAD

**Discussion by mailys-ds** [microsoft/pyright#3865] [Q&A] [answered]
Hello,

It seems that since 1.1.242, pyright doesn't detect circular import in chains that are made up of more than 2 modules.

To reproduce:
``` python
# module_a.py
from .module_b import B
class A():
    def foo(self, b: B) -> B:
        return b

# module_b.py
from .module_c import C
class B():
    def foo(self, c: C) -> C:
        return c

# module_c.py
from .module_a import A
class C():
    def foo(self, a: A) -> A:
        return a
```
This circular import used to be detected in pyright < 1.1.242. So is this behaviour expected or is it a bug ?

Thanks,
Maïlys.

**Comment by erictraut** [comment, marked as answer]
When I enable `reportImportCycles` and run it on the above code, I see the expected diagnostic `Cycle detected in import chain`, and it lists the three modules in the cycle. Is there anything else I need to do to repro what you're seeing?

**Reply by mailys-ds** [reply]
Thanks for your quick answer. Here is more details to reproduce my issue:
I created a folder test_pyright with the following structure:
```
test_pyright/
        __init__.py
        module_a.py
        module_b.py
        module_c.py
```
I enable `reportImportCycles` with  `reportImportCycles= "warning"` in my `pyproject.toml`  and run `pyright ./test_pyright` on Linux with python 3.8.12 and pyright 1.1.268, no warning is raised.
Whereas, if I run the same command with pyright 1.1.241, the warning `Cycle detected in import chain` with the three modules is listed as expected.

Let me know if you manage to reproduce it.

**Reply by erictraut** [reply]
Thanks, I'm able to repro the problem with the CLI. The error is reported in the language server in VS Code. I need to determine why I'm seeing a difference between the two.

Converting the discussion into a bug report.
